src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const copyState = Object.assign({}, state);
+  const copyStateArray = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        Object.keys(copyState).forEach(key => {
+          delete copyState[key];
```

Comment:
it's better to use 'for in' in this case

Reply:
+